FEATURE - HISTORY - Add & Get conversions favorites

Deivid117 opened the pull request:
- Se agregó la lógica para marcar una conversión como favorita
- Se agregó la lógica para obtener todas las conversiones marcadas como favoritas
- También se aumentó en 2 puntos el tamaño de la fuente bodyLarge

[Tareas](https://trello.com/c/DXfSsWQa)

Deivid117 committed:
FEATURE - Add conversions to favorites

- Se agregó la lógica para marcar como favorita una conversión
- Creación del repository, use case, implementación y query en el DAO

Deivid117 committed:
Merge branch 'develop' of https://github.com/VarquLabs/DolarBlueApp into feature/add-conversions-favorite

Deivid117 committed:
FEATURE - Get favorite conversion history

- Se realizó la lógica para obtener todas las conversiones favoritas en el historial
- Se hizo el query en el dao, la función del repository y la implementación

Deivid117 committed:
FEATURE - Get favorite conversion history use case

- Se agregó el caso de uso para las conversiones favoritas

Deivid117 committed:
FEATURE - Provide use case

- Se agregó el provide del use case al repository module

Deivid117 committed:
REFACTOR - Updated query for get favorite conversions

- Se modificó el query para que obtenga primero toda la información de la tabla de conversiones

Deivid117 committed:
REFACTOR - Grouping and mapping function

- Se agregó un función para agrupar y mapear las respuestas de las funciones del dao de obtener los favoritos y buscar conversiones para no repetir código

Deivid117 committed:
Merge branch 'develop' of https://github.com/VarquLabs/DolarBlueApp into feature/add-conversions-favorite

Deivid117 committed:
REFACTOR - Size change for bodyLarge

- Se aumentó 2 puntos el tamaño de la fuente para bodyLarge

juanjosevarqu commented on app/src/main/java/com/varqulabs/dolarblue/history/data/repository/ConversionsHistoryRepositoryImpl.kt:
```diff
@@ -16,8 +17,20 @@ class ConversionsHistoryRepositoryImpl(
+        conversionHistoryDao.addConversionFavorite(conversionId, isFavorite)
+    }
+
+    private fun groupByExchangeRateIdAndMapToConversionHistory(daoFunction: Flow<List<ConversionsWithCurrentExchangeRelation>>) : Flow<List<ConversionsHistory>> {
```
Podemos buscar un nombre mas simple para esta funcion

El "groupBy" lo hace dentro, pero puede que en el futuro encontremos otra forma. O directo en el Query

Deivid117 replied:
¿Alguna otra propuesta? No se me ocurrió otra idea para el nombre?

juanjosevarqu commented on app/src/main/java/com/varqulabs/dolarblue/history/data/local/database/dao/ConversionsHistoryDao.kt:
```diff
@@ -18,13 +18,25 @@ interface ConversionsHistoryDao {
     fun searchConversionsHistoryByQuery(querySearch: String): Flow<List<ConversionsWithCurrentExchangeRelation>>
+
+    @Query("UPDATE conversion_table SET isFavorite = :isFavorite WHERE id = :conversionId")
+    fun addConversionFavorite(conversionId: Int, isFavorite: Boolean)
```
Podria llamarse addConversionToFavorites

juanjosevarqu commented on app/src/main/java/com/varqulabs/dolarblue/history/domain/useCases/AddConversionFavoriteUseCase.kt:
```diff
@@ -0,0 +1,11 @@
+
+import com.varqulabs.dolarblue.history.domain.repository.ConversionsHistoryRepository
+
+class AddConversionFavoriteUseCase(
```
X2

juanjosevarqu commented on app/src/main/java/com/varqulabs/dolarblue/history/domain/useCases/AddConversionFavoriteUseCase.kt:
```diff
@@ -0,0 +1,11 @@
+class AddConversionFavoriteUseCase(
+    private val repository: ConversionsHistoryRepository
+) {
+    suspend operator fun invoke(conversionId: Int, isFavorite: Boolean) {
```
Es necesario el parametro isFavorite ?
El Caso de Uso da a entender que se agregara a Favoritos

juanjosevarqu replied:
La Validación de si es Favorito o no para llamar al caso de uso, puede hacerse en el ViewModel

Deivid117 replied:
Con ese mismo pretendo que sirva tanto para marcar como favorito como para quitarlo de favoritos en un solo caso de uso, por eso le envío ese parámetro, no sé si pueda ir otro nombre para ambas acciones

juanjosevarqu reviewed (changes requested):
Deje una review

Deivid117 committed:
REFACTOR - Update conversion

- Se reemplazó la función del dao para cambiar el nombre de una conversión por un query de Update.
- Se refactorizaron las funciones de addConversionFavorite por updateConversion.
- El caso de uso cambió para que en cualquier caso actualice cualquer parámetro que se el envíe de la conversion.
- Se agregó un mapper para convertir el modelo de conversion por una entity
- Se cambió el nombre de la función que agrupa las conversiones en repository impl

juanjosevarqu commented on app/src/main/java/com/varqulabs/dolarblue/history/domain/mappers/ConversionMapper.kt:
```diff
@@ -0,0 +1,8 @@
+import com.varqulabs.dolarblue.calculator.data.local.database.entities.ConversionEntity
+import com.varqulabs.dolarblue.history.domain.model.Conversion
+
+fun Conversion.mapToEntity() = ConversionEntity(
```
Reformat y nombre de los parametros

Deivid117 committed:
REFACTOR - Conversion model

- Se agregó el nombre de los parámetros
